Content.Client/UserInterface/AdminMenu/AdminMenuWindow.cs

```diff
@@ -0,0 +1,615 @@
+                    IoCManager.Resolve<IPrototypeManager>(),
+                    IoCManager.Resolve<IResourceCache>(),
+                    IoCManager.Resolve<ILocalizationManager>());
+                manager.OpenCommand(window);
+            }
+        }
+
+        private class SpawnTilesCommandButton : CommandButton
+        {
+            public override string Name => "Spawn Tiles";
+            //TODO: override CanPress
+            public override void ButtonPressed(ButtonEventArgs args)
+            {
+                var manager = IoCManager.Resolve<IAdminMenuManager>();
+                var window = new TileSpawnWindow(IoCManager.Resolve<ITileDefinitionManager>(),
+                    IoCManager.Resolve<IPlacementManager>(),
+                    IoCManager.Resolve<IResourceCache>());
+                manager.OpenCommand(window);
+            }
+        }
+
+        private class KickCommandButton : UICommandButton
+        {
+            public override string Name => "Kick";
+            public override string RequiredCommand => "kick";
+
+            public override List<CommandUIControl> UI => new List<CommandUIControl>
+            {
+                new CommandUIDropDown
+                {
+                    Name = "Player",
+                    GetData = () => IoCManager.Resolve<IPlayerManager>().Sessions.ToList<object>(),
+                    GetDisplayName = (obj) => $"{((IPlayerSession)obj).Name} ({((IPlayerSession)obj).AttachedEntity?.Name})",
+                    GetValueFromData = (obj) => ((IPlayerSession)obj).Name,
+                },
+                new CommandUILineEdit
+                {
+                    Name = "Reason"
+                }
+            };
+
+            public override void Submit(Dictionary<string,string> val)
+            {
+                IoCManager.Resolve<IClientConsole>().ProcessCommand($"kick \"{val["Player"]}\" \"{val["Reason"].Replace("\\", "\\\\").Replace("\"", "\\\"")}\"");
+            }
+        }
+
+        private class TestCommandButton : UICommandButton
+        {
+            public override string Name => "Test";
+
+            public override List<CommandUIControl> UI => new List<CommandUIControl>
+            {
+                new CommandUIDropDown
+                {
+                    Name = "DropDown",
+                    GetData = () => new List<object>
+                    {
+                        "1",
+                        "2"
+                    },
+                    GetDisplayName = (obj) => (string)obj,
+                    GetValueFromData = (obj) => (string)obj
+                },
+                new CommandUILineEdit
+                {
+                    Name = "LineEdit"
+                },
+                new CommandUICheckBox
+                {
+                    Name = "CheckBox"
+                },
+                new CommandUISpinBox
+                {
+                    Name = "SpinBox"
+                },
```

Comment:
I could do that to make the code cleaner, but the Command thingies don't really hold data itself (except the dropdown). The only thing that this could do is remove the need for 2 Name strings (in the submit & the definition)